source/opt/dead_branch_elim_pass.cpp

```diff
@@ -331,8 +331,8 @@ bool DeadBranchElimPass::EliminateDeadBranches(ir::Function* func) {
       }
       const uint32_t phiId = pii->result_id();
       KillNamesAndDecorates(phiId);
-      (void)get_def_use_mgr()->ReplaceAllUsesWith(phiId, replId);
-      get_def_use_mgr()->KillInst(&*pii);
+      (void)context()->ReplaceAllUsesWith(phiId, replId);
+      context()->KillInst(&*pii);
```

Comment:
Out of curiosity.  What was the rationale you used to move some functions into IRContext vs the DefUse manager?  Is it instruction manipulation that goes in IRContext now?  And pure analysis stays in DefUse Manager?

Reply:
Yes, functions that change the IR are part of the IRContext.  This way the def-use manager will not have to be aware of other analyses that are added in the future.

Reply:
Sounds good.